File: SelectionSort/SelectionSort.java
import org.jetbrains.annotations.NotNull;

public class SelectionSort {
    public static void selectionSort(int @NotNull [] array) {
        for(int i = 0; i < array.length; i++) {
            int minimumIndex = i;
            for(int j = i; j < array.length; j++) {
                if(array[j] < array[minimumIndex]) {
                    minimumIndex = j;
                }
            }
            int temp = array[i];
            array[i] = array[minimumIndex];
            array[minimumIndex] = temp;
        }
    }

    public static void printArray(String arrayName, int @NotNull [] array) {
        System.out.print(arrayName + " = { ");
        for(int i = 0; i < array.length; i++) {
            System.out.print(array[i]);
            if(i != array.length - 1) System.out.print(", ");
        }
        System.out.println(" }");
    }

    public static void main(String[] args) {
        int[] numberArray = {6, 62, 14, 96, 69, 67, 32, 85};
        printArray("UNSORTED ARRAY", numberArray);
        // UNSORTED ARRAY = { 6, 62, 14, 96, 69, 67, 32, 85 }
        selectionSort(numberArray);
        printArray("SORTED ARRAY", numberArray);
        // SORTED ARRAY = { 6, 14, 32, 62, 67, 69, 85, 96 }
    }
}

File: SelectionSort/selectionSort.c
#include <stdio.h>

void selectionSort(int array[], int arraySize) {
    for(int i = 0; i < arraySize; i++) {
        int minimumIndex = i;
        for(int j = i; j < arraySize; j++) {
            if(array[j] < array[minimumIndex]) {
                minimumIndex = j;
            }
        }
        int temp = array[i];
        array[i] = array[minimumIndex];
        array[minimumIndex] = temp;
    }
}

void printArray(char arrayName[], int array[], int arraySize) {
    printf("%s = { ", arrayName);
    for(int i = 0; i < arraySize; i++) {
        printf("%d", array[i]);
        if(i != arraySize - 1) printf(", ");
    }
    printf(" }\n");
}

int main() {
    int numberArray[] = {55, 33, 36, 87, 40, 7, 49, 4};
    int arraySize = sizeof(numberArray) / sizeof(int);
    printArray("UNSORTED ARRAY", numberArray, arraySize);
    // UNSORTED ARRAY = { 55, 33, 36, 87, 40, 7, 49, 4 }
    selectionSort(numberArray, arraySize);
    printArray("SORTED ARRAY", numberArray, arraySize);
    // SORTED ARRAY = { 4, 7, 33, 36, 40, 49, 55, 87 }
    return 0;
}

